core/isolate.rs

```diff
@@ -315,6 +315,21 @@ impl Isolate {
       PinnedBuf::new(zero_copy_buf),
     );
 
+    let op = match op {
+      Op::Async(mut fut) => {
+        // Tries to greedily poll async ops once. Often they are immediately ready, in
+        // which case they can be turned into a sync op before we return to V8. This
+        // can save a boundary crossing.
+        #[allow(clippy::match_wild_err_arm)]
+        match fut.poll() {
+          Err(_) => panic!("unexpected op error"),
+          Ok(Ready(buf)) => Op::Sync(buf),
+          Ok(NotReady) => Op::Async(fut),
+        }
+      }
+      Op::Sync(buf) => Op::Sync(buf),
+    };
+
```

Comment:
Critical change in this PR

Reply:
It's not working properly ATM. I'm `polling` on a future outside of task context

Reply:
Can you undo the changes in the http_bench code and minimize the patch so we're only dealing with the greedy bit?

This poll is actually inside a future task context, since it's called inside Isolate::poll.

Reply:
@ry I reverted the changes to `http_bench`.

This function is `pre_dispatch` which I believe is called directly by V8 using `recv_cb`